tsdb/store.go

```diff
@@ -351,19 +352,19 @@ func (s *Store) loadShards() error {
 				return err
 			}
 
-			for _, sh := range shardDirs {
+			for index, sh := range shardDirs {
 				// Series file should not be in a retention policy but skip just in case.
 				if sh.Name() == SeriesFileDirectory {
 					log.Warn("Skipping series file in retention policy dir", zap.String("path", filepath.Join(s.path, db.Name(), rp.Name())))
 					continue
 				}
 
 				n++
-				go func(db, rp, sh string) {
+				go func(db, rp, sh string, id int) {
```

Comment:
are we relying on this function closing over any variables?  if not, can we make this a separate function?